src/types/constraint.ts

```diff
@@ -23,11 +23,12 @@ export type ConstraintViolation = {
   constraintId: Constraint['id'];
   constraintName: Constraint['name'];
   gaps: TimeRange[];
+  type: ConstraintType;
   windows: TimeRange[];
 };
 
 export type ConstraintViolationsMap = Record<Constraint['id'], ConstraintViolation[]>;
 
-export type ConstraintViolationsResponseMap = Record<Constraint['name'], Omit<ConstraintViolation, 'constraintName'>[]>;
+export type ConstraintViolationsResponse = ConstraintViolation[];
```

Comment:
Can we just remove this type now and type the response as `ConstraintViolation[]` where applicable?

Reply:
Yeah, good idea.